openpype/plugins/publish/integrate_version_attrs.py

```diff
@@ -0,0 +1,92 @@
+import pyblish.api
+import ayon_api
+
+from openpype import AYON_SERVER_ENABLED
+from openpype.client.operations import OperationsSession
+
+
+class IntegrateVersionAttributes(pyblish.api.ContextPlugin):
+    """Integrate version attributes from predefined key.
+
+    Any integration after 'IntegrateAsset' can fill 'versionAttributes' with
+    attribute key & value to be updated on created version.
+
+    The integration must make sure the attribute is available for the version
+    entity otherwise an error would be raised.
+
+    Example of 'versionAttributes':
+        {
+            "ftrack_id": "0123456789-101112-131415",
+            "syncsketch_id": "987654321-012345-678910"
+        }
+    """
+
+    label = "Integrate Version Attributes"
+    order = pyblish.api.IntegratorOrder + 0.5
+
+    def process(self, context):
+        available_attributes = ayon_api.get_attributes_for_type("version")
+        skipped_attributes = set()
+        project_name = context.data["projectName"]
+        op_session = OperationsSession()
+        for instance in context:
+            label = self.get_instance_label(instance)
+            version_entity = instance.data.get("versionEntity")
+            if not version_entity:
+                continue
+            attributes = instance.data.get("versionAttributes")
+            if not attributes:
+                self.log.debug((
+                    "Skipping instance {} because it does not have"
+                    " specified version attributes to set."
+                ).format(label))
+                continue
+
+            filtered_attributes = {}
+            for attr, value in attributes.items():
+                if attr not in available_attributes:
+                    skipped_attributes.add(attr)
+                else:
+                    filtered_attributes[attr] = value
+
+            if not filtered_attributes:
+                self.log.debug((
+                    "Skipping instance {} all attributes were filtered out."
+                ).format(label))
+                continue
+
+            self.log.debug("Updating attributes on version {} to {}".format(
+                version_entity["_id"], str(filtered_attributes)
+            ))
+            op_session.update_entity(
+                project_name,
+                "version",
+                version_entity["_id"],
+                {"attrib": filtered_attributes}
+            )
+
+        if skipped_attributes:
+            self.log.warning((
+                "Skipped attributes because they're not available"
+                " on the server: {}"
+            ).format(str(skipped_attributes)))
```

Comment:
I still feel this message can be improved. By just reading the log it's unclear to me where they are not available on the server. What does it mean if they are not available? Etc. With an effort to improve logged messages (with this one for example being visible to an artist) it'd be nice if these could point as close as possible to the source of the problem).

This at least mentions what attributes and where it's missing:
```suggestion
            self.log.warning((
                "Skipped version attributes because they're not available"
                " on the server for version entities: {}"
            ).format(str(skipped_attributes)))
```
Still quite technical, but least provides context to what it's referring to?